### PR TITLE
Apply generator-videojs-plugin conventions

### LINKED ISSUE #8: Double add overlays.
In my testing I have something like:

```
player.overlay({
  overlays: [{
    content: "Test1",
    start: 0,
    end: 15,
    align: 'bottom'
   }, 
   {
    content: "Test2",
    start: 17,
    end: 30,
    align: 'bottom'
  }]
});
```

If I click more than once on the scrubber at around 12 seconds for example,  I get multiple overlays displayed. Which then do not remove after the end trigger.

### PR DESCRIPTION
**Pull request by misteroneill**
In addition to the general update to the codebase to follow the generator conventions, this fixes #8 and a few other minor/unreported bugs.

### CODE REVIEW

**Comment by gkatsev** [comment]
Not really familiar with this module. However, LGTM. Tests are still broken though.

File: .travis.yml [hunk 1]
```diff
@@ -1,6 +1,13 @@
+sudo: false
 language: node_js
 node_js:
-- '0.10'
+  - 'node'
+  - '4.2'
```
Comment (gkatsev) [inline .travis.yml]: this reminds me, we should update this to 4.3 as that's now the latest LTS version.

File: LICENSE [hunk 2]
```diff
@@ -1,4 +1,4 @@
-Copyright 2013 Brightcove
+Copyright 2016 Brightcove, Inc.
```
Comment (gkatsev) [inline LICENSE]: Just delete the year :)

File: src/plugin.js [hunk 3]
```diff
@@ -0,0 +1,317 @@
...
+    let log = videojs.log;
+    let fn = log;
+
+    // Support `videojs.log.foo` calls.
+    if (log.hasOwnProperty(args[0]) && typeof log[args[0]] === 'function') {
+      fn = log[args.shift()];
+    }
+
+    args.unshift(`overlay#${this.id()}: `);
+    fn(...args);
```
Comment (gkatsev) [inline src/plugin.js]:
```js
fn(...[`overlay#${this.id)`, ...args]);
```
but yeah, not necessary

File: .travis.yml (hunk 1 above)
Comment (misteroneill) [inline .travis.yml]: I'll create an issue for the generator.

File: LICENSE (hunk 2 above)
Comment (misteroneill) [inline LICENSE]: :+1: I'll create an issue for the generator.

File: src/plugin.js (hunk 3 above)
Comment (misteroneill) [inline src/plugin.js]: Ah, yes, good call. I'll include this change as well. Might as well, while I'm in there.

File: src/plugin.js [hunk 4]
```diff
@@ -0,0 +1,317 @@
...
+ *
+ * @class   Overlay
+ * @extends {videojs.Component}
+ */
+class Overlay extends Component {
+
+  constructor(player, options) {
+    super(player, options);
+
+    player.addChild(this);
```
Comment (gkatsev) [inline src/plugin.js]: this seems odd that the component adds itself to the player.

File: src/plugin.js [hunk 5]
```diff
@@ -0,0 +1,317 @@
...
+  }
+
+  const overlays = settings.overlays;
+
+  // We don't want to keep the original array of overlay options around
+  // because it doesn't make sense to pass it to each Overlay component.
+  delete settings.overlays;
+
+  this.overlays_ = overlays.map(
+    o => new Overlay(this, videojs.mergeOptions(settings, o))
```
Comment (gkatsev) [inline src/plugin.js]: perhaps this should be `player.addChild(Overlay, videojs.mergeOptions(settings, o))`, I believe that should work.

File: src/plugin.js (hunk 4 above)
Comment (misteroneill) [inline src/plugin.js]: That's a fair point. I'll move it out into the plugin function. :+1:

File: src/plugin.js (hunk 5 above)
Comment (misteroneill) [inline src/plugin.js]: Looks like you have to instantiate it first, which is fine.

Comment (gkatsev) [inline src/plugin.js]: actually, given that it's registered as a component, you could just do `player.addChild('overlay')`

File: src/plugin.js [hunk 6]
```diff
@@ -0,0 +1,314 @@
...
+  }
+
+  const overlays = settings.overlays;
+
+  // We don't want to keep the original array of overlay options around
+  // because it doesn't make sense to pass it to each Overlay component.
+  delete settings.overlays;
+
+  this.overlays_ = overlays.map(o => {
+    this.addChild('Overlay', videojs.mergeOptions(settings, o));
```
Comment (gkatsev) [inline src/plugin.js]: I believe it has to be lowercase.
Also, this shouldn't be wrapped in `{}` should it?

Comment (misteroneill) [inline src/plugin.js]: It works as-is, but you're right - the `{}` is unnecessary.

Comment (gkatsev) [inline src/plugin.js]: Well, the `{}` change the meaning of the function. I guess it works because `addChild` just upper cases the first letter.

Comment (misteroneill) [inline src/plugin.js]: True. I didn't think about that fact that it's `map`ping. The weekend can't come soon enough! :smile:

Comment (gkatsev) [inline src/plugin.js]: That is, if we want `this.overlays_` to be an array of the overlays, we have to remove the `{}` or add `return`.

Comment (misteroneill) [inline src/plugin.js]: Yes, that's what I did. The intent was not an array of `undefined`s!

**Comment by gkatsev** [comment]
LGTM

**Pull request merged**